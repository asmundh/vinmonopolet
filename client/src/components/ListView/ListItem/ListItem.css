.item-icon {
    width: 100%;
}