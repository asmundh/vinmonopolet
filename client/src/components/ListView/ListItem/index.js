import React, { Component } from 'react';
import { Table, Icon } from 'semantic-ui-react';
import './ListItem.css';

class ListItem extends Component {

    // Bindings
    constructor() {
        super();
        this.handleChange = this.handleChange.bind(this);
        this.handleRating = this.handleRating.bind(this);
    }

    // Handler for user selecting an item
    handleChange = () => {
        this.props.onClick(this.props.index);
    };

    // Handler for user giving a rating (like/dislike)
    handleRating = (isLike) =>
        this.props.handleRating(this.props.index, isLike);

    // Returns appropriate icon-type based on keywords in item type.
    getIconType = () => {
        if(this.props.type.toLowerCase().includes("vin")) return "wine";
        if(this.props.type.toLowerCase().includes("cha")) return "wine";
        else if(this.props.type.toLowerCase().includes("Ã¸l")) return "beer";
        else if(this.props.type.toLowerCase().includes("ale")) return "beer";
        else if(this.props.type.toLowerCase().includes("lager")) return "beer";
        else if(this.props.type.toLowerCase().includes("saison")) return "beer";
        else return "liquor";
    };

    render() {
        return (
                <Table.Row className='db-item' style={{cursor: "pointer"}}>
                    <Table.Cell className='db-item-field-image'
                                onClick={() => this.handleChange()} textAlign={"center"}>
                        <div className={"image-container"}>
                            <img className="item-icon" src={require(`../../../resources/${this.getIconType()}icon.png`)}
                                alt={"Icon"} />
                        </div>
                    </Table.Cell>
                    <Table.Cell className='db-item-field-name' onClick={() => this.handleChange()} >{this.props.name}</Table.Cell>
                    <Table.Cell className='db-item-field-type' onClick={() => this.handleChange()} >{this.props.type}</Table.Cell>
                    <Table.Cell className='db-item-field-alcohol' onClick={() => this.handleChange()} >{this.props.alcohol}</Table.Cell>
                    <Table.Cell className='db-item-field-volume' onClick={() => this.handleChange()} >{this.props.volume}</Table.Cell>
                    <Table.Cell className='db-item-field-price' onClick={() => this.handleChange()} >{this.props.price}</Table.Cell>
                    <Table.Cell className='db-item-field-country' onClick={() => this.handleChange()} >{this.props.country}</Table.Cell>
                    <Table.Cell className='db-item-field-apk' onClick={() => this.handleChange()}>
                        <b style={this.props.apk < 0.05 ? {color: `rgba(206, 58, 28, ${(1-this.props.apk*10)})`} :
                            {color: `rgba(89, 204, 22, ${this.props.apk*10})`}}>
                        {Math.round(this.props.apk*1000)/1000}</b>
                    </Table.Cell>
                    <Table.Cell className='db-item-field-rating' textAlign={'center'}>
                        <div style={{overflow: "hidden"}}>
                            <Icon name={"thumbs down outline"} id="thumbs-down" onClick={() => this.handleRating(false)}/>
                            <b id="score" style={
                                (this.props.likes-this.props.dislikes) < 0 ? {color: "#ce3a1c"} : {color: "#59cc16"}
                            }>{(this.props.likes-this.props.dislikes)}</b>
                            <Icon name={"thumbs up outline"} id="thumbs-up" onClick={() => this.handleRating(true)}
                                  style={{marginLeft: "5px"}}/>
                        </div>
                    </Table.Cell>
                </Table.Row>
        );
    }
}

export default ListItem;
