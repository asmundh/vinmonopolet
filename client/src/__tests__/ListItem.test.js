import React from 'react';
import ListItem from "../components/ListView/ListItem";
import renderer from 'react-test-renderer';

    // Snapshot-test
test('snapshot-test: vin', ()=> {
   const tree = renderer.create(<ListItem type={"vin"}/>).toJSON();
   expect(tree).toMatchSnapshot();
});

   // Snapshot-test
test('renders correctly: øl', ()=> {
   const tree = renderer.create(<ListItem type={"øl"}/>).toJSON();
   expect(tree).toMatchSnapshot();
});

  // Snapshot-test
test('renders correctly: lager', ()=> {
   const tree = renderer.create(<ListItem type={"lager"}/>).toJSON();
   expect(tree).toMatchSnapshot();
});