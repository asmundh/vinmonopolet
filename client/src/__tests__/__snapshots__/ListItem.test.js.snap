// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly: lager 1`] = `
<tr
  className="db-item"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <td
    className="center aligned db-item-field-image"
    onClick={[Function]}
  >
    <div
      className="image-container"
    >
      <img
        alt="Icon"
        className="item-icon"
        src={require("../public/resources/beericon.png")}
      />
    </div>
  </td>
  <td
    className="db-item-field-name"
    onClick={[Function]}
  />
  <td
    className="db-item-field-type"
    onClick={[Function]}
  >
    lager
  </td>
  <td
    className="db-item-field-alcohol"
    onClick={[Function]}
  />
  <td
    className="db-item-field-volume"
    onClick={[Function]}
  />
  <td
    className="db-item-field-price"
    onClick={[Function]}
  />
  <td
    className="db-item-field-country"
    onClick={[Function]}
  />
  <td
    className="db-item-field-apk"
    onClick={[Function]}
  >
    <b
      style={
        Object {
          "color": "rgba(89, 204, 22, NaN)",
        }
      }
    >
      NaN
    </b>
  </td>
  <td
    className="center aligned db-item-field-rating"
  >
    <div
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <i
        aria-hidden="true"
        className="thumbs down outline icon"
        onClick={[Function]}
      />
      <b
        style={
          Object {
            "color": "#59cc16",
          }
        }
      >
        NaN
      </b>
      <i
        aria-hidden="true"
        className="thumbs up outline icon"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "5px",
          }
        }
      />
    </div>
  </td>
</tr>
`;

exports[`renders correctly: øl 1`] = `
<tr
  className="db-item"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <td
    className="center aligned db-item-field-image"
    onClick={[Function]}
  >
    <div
      className="image-container"
    >
      <img
        alt="Icon"
        className="item-icon"
        src={require("../public/resources/beericon.png")}
      />
    </div>
  </td>
  <td
    className="db-item-field-name"
    onClick={[Function]}
  />
  <td
    className="db-item-field-type"
    onClick={[Function]}
  >
    øl
  </td>
  <td
    className="db-item-field-alcohol"
    onClick={[Function]}
  />
  <td
    className="db-item-field-volume"
    onClick={[Function]}
  />
  <td
    className="db-item-field-price"
    onClick={[Function]}
  />
  <td
    className="db-item-field-country"
    onClick={[Function]}
  />
  <td
    className="db-item-field-apk"
    onClick={[Function]}
  >
    <b
      style={
        Object {
          "color": "rgba(89, 204, 22, NaN)",
        }
      }
    >
      NaN
    </b>
  </td>
  <td
    className="center aligned db-item-field-rating"
  >
    <div
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <i
        aria-hidden="true"
        className="thumbs down outline icon"
        onClick={[Function]}
      />
      <b
        style={
          Object {
            "color": "#59cc16",
          }
        }
      >
        NaN
      </b>
      <i
        aria-hidden="true"
        className="thumbs up outline icon"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "5px",
          }
        }
      />
    </div>
  </td>
</tr>
`;

exports[`snapshot-test: vin 1`] = `
<tr
  className="db-item"
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <td
    className="center aligned db-item-field-image"
    onClick={[Function]}
  >
    <div
      className="image-container"
    >
      <img
        alt="Icon"
        className="item-icon"
        src={require("../public/resources/wineicon.png")}
      />
    </div>
  </td>
  <td
    className="db-item-field-name"
    onClick={[Function]}
  />
  <td
    className="db-item-field-type"
    onClick={[Function]}
  >
    vin
  </td>
  <td
    className="db-item-field-alcohol"
    onClick={[Function]}
  />
  <td
    className="db-item-field-volume"
    onClick={[Function]}
  />
  <td
    className="db-item-field-price"
    onClick={[Function]}
  />
  <td
    className="db-item-field-country"
    onClick={[Function]}
  />
  <td
    className="db-item-field-apk"
    onClick={[Function]}
  >
    <b
      style={
        Object {
          "color": "rgba(89, 204, 22, NaN)",
        }
      }
    >
      NaN
    </b>
  </td>
  <td
    className="center aligned db-item-field-rating"
  >
    <div
      style={
        Object {
          "overflow": "hidden",
        }
      }
    >
      <i
        aria-hidden="true"
        className="thumbs down outline icon"
        onClick={[Function]}
      />
      <b
        style={
          Object {
            "color": "#59cc16",
          }
        }
      >
        NaN
      </b>
      <i
        aria-hidden="true"
        className="thumbs up outline icon"
        onClick={[Function]}
        style={
          Object {
            "marginLeft": "5px",
          }
        }
      />
    </div>
  </td>
</tr>
`;
